webapp/src/Controller/PublicController.php

```diff
@@ -267,4 +282,54 @@ protected function getBinaryFile(int $probId, callable $response): StreamedRespo
 
         return $response($probId, $contest, $contestProblem);
     }
+
+    #[Route(path: '/submissions/team/{teamId<\d+>}/problem/{problemId<\d+>}', name: 'public_submissions')]
+    public function submissionsAction(Request $request, int $teamId, int $problemId): Response
+    {
+        $contest = $this->dj->getCurrentContest(onlyPublic: true);
+
+        if (!$contest) {
+            throw $this->createNotFoundException('No active contest found');
+        }
+
+        /** @var Team|null $team */
+        $team             = $this->em->getRepository(Team::class)->find($teamId);
+        if ($team && $team->getCategory() && !$team->getCategory()->getVisible()) {
+            $team = null;
+        }
+
+        if (!$team) {
+            throw $this->createNotFoundException('Team not found');
```

Comment:
All errors get better with another `.`.
```suggestion
            throw $this->createNotFoundException('Team not found.');
```